extensions/2.0/Khronos/KHR_image_formation/README.md

```diff
@@ -0,0 +1,197 @@
+
+The non-linear LUT is much more complicated and arbitrary, but is nevertheless completely necessary. This compression step is necessary because the [0, infinity) range of physical light must be converted to the [0, 1] range of any output format. When no LUT is specified, this generally means the value is clamped, which is simply a piecewise linear compression function with sharp corners that  create unsightly artifacts. This is rarely the default mode for existing renderers, as it yields such poor results, but their  defaults are also rarely consistent with each other, hence the value of this extension.
+
+There is no “right” way to compress outgoing scene light, but the goal is to mimic the nonlinear response of the human retina without a lot of expensive computation. Camera sensors and film are likewise nonlinear and designed also around the human eye’s response. This is the purpose of the hdr_lut. Effectively we are mapping from physical light units to unitless perception values, represented as sRGB or other wide-gamut, HDR formats. 
+
+One key aspect of a good transform is that it does not operate solely on luminance but also affects color, especially saturation. This is because as a colored light becomes very bright, its color ceases to be perceived. This is true of CCDs as well, and can be easily seen by taking an overexposed photo of a bright, colored light and noting how it tends toward white. This is the purpose of the tone_mapping_lut.
+
+Regarding color gamuts, since glTF defines its textures in sRGB, so long as the scene’s lighting is also represented in Rec 709 (sRGB’s color gamut), then the output light will naturally also be restricted to the Rec 709 gamut. If a wide gamut like Rec 2020 is used for output, then the values will naturally fall into the Rec 709 subset. 
+
+## Extending Scene
+
+The precise method of converting scene linear light into output pixels is defined by adding the `KHR_image_formation` extension to any glTF scene. For example:
+
+```json
+{
+    "scene": [
+        {
+            "extensions": {
+                "KHR_image_formation": {
+                    "exposure": 1.0,
+                    "hdr_lut": "hdr.cube",
+                    "tone_mapping_lut": "tone.cube",
+                }
+            }
+        }
+    ]
+}
+```
+
+| Property | Description | Required |
+|:-----------------------|:------------------------------------------| :--------------------------|
+| `exposure` | Linear multiplier on lighting. Must be a positive value. | No, Default: `1.0` |
+| `hdr_lut` | Link to a 1D .cube file, defining the first post-processing step. | No |
```

Comment:
I think that adding the option of a LUT to a glTF changes the PBR nature of the specification.
As I see it we are aiming towards a physically correct result with the glTF format.
A LUT may radically change the color (hue) coming out of the glTF BRDF, breaking the physical correctness.
How do you see this 'artistic intent' opposed to physical correctness?

Reply:
No, a LUT is part of the output process of a physical camera creating an image (and I would argue a very similar process happens in the eye's perception of light, though that is far more complex). So it is equally a part of physically-based rendering, as long as the goal is output to sRGB pixel values.

Reply:
This is meant to be flexible, allow *both* "neutral" LUTs like product photographers use, as well as artistic ones like Sepia if desired. Again, this is just an extension, and can be easily ignored by a consumer that it doesn't make sense for.

Reply:
A LUT does not change the physical correctness of the materials definition and lighting model, it does however define image formation. Those are two separate and not incompatible steps.

Reply:
>No, a LUT is part of the output process of a physical camera creating an image

If that is the case I would argue that the LUT is applied as a last step when consuming the image, not as part of the asset itself.
Which leads me to thinking that a (physical) camera extension would be a better fit for your usecase.

Reply:
> This is meant to be flexible, allowing _both_ "neutral" LUTs like product photographers use, as well as artistic ones like Sepia if desired.

I see this as a process that belongs at the consumer end, ie in the renderer or engine.
 - Similar to when Instagram filters are applied by the app.

Reply:
> Again, this is just an extension, and can be easily ignored by a consumer that it doesn't make sense for.

I believe this extension, if used, has much broader impliclations.
Imagine the impact for a company that has 10K + models - all models need settings to get the intended look and feel.
This will be very error prone and may lead to problems in content creation pipeline. 
For instance, the same model authored by different persons may have a different look.

I advice against having exposure and LUT in the asset.

Reply:
> A LUT does not change the physical correctness of the materials definition and lighting model, it does however define image formation.

In that case I think it is important to state that while the glTF (internal) BRDF calculations are unaffected - the output of this extension may not be physically correct.
Would you agree to that statement?

Reply:
> In that case I think it is important to state that while the glTF (internal) BRDF calculations are unaffected - the output of this extension may not be physically correct.

What is physically correct at the display? I would argue that it could only be considered physically correct given a perfect, known display at a known sustained peak brightness, and in a known, fixed viewing environment (and this means that the viewer should see only the glTF scene and nothing else to avoid any kind of adaptation). What this extension tries to achieve is in its very name: it's about image formation, and therefore perception rather than stimulus. And chasing after "physical correctness" at the display level is, imo, foolish considering the viewing environments (on and outside the display), display settings, display capabilities, etc.

A very simple and telling example to me is the game I was playing last night, Gran Turismo 7. It outputs HDR and is clearly not "physically correct" at the display (since it shows the sun in the sky and it's certainly not outputting 100,000 lux), but is the *perception* of the scene realistic/believable? Absolutely.

Reply:
> What this extension tries to achieve is in its very name: it's about image formation, and therefore perception rather than stimulus.

Maybe this is what makes it non physically correct in my opinion.
Allowing the perception of a glTF to be monochromatic (black&white) - when the surface properties and light environment is chromatic (colored) - is not what I would call physically correct.

My definition of "physically correct output" does not mean that it must be a 100% replication of light intensity levels 'as they are'
The contrast between higher intensity and lower intensity light shall be as correct as possible when output.

In the case of a brightly lit summer day, let's say 100 000 lumen / m2.
The amount of light that will enter through the iris and pupil, hitting the photoreceptor cells on the retina is much, much smaller. 

Most of it will be blocked by the constricted pupil to protect the photoreceptor cells.
This means that light as 'captured' by the observer is in a smaller range.
Let's say that of the sun's 100 000 lumen/m2, only 10 000 lumen/m2 will enter into the eye.
The same thing will happen to the light coming from other parts of the environment making them look dark (since you are looking towards a very bright spot)

It is this contrast, between the bright sky and the shaded areas, that I would like to get to the display as it would be 'captured' by the observer. 
(In this case, shaded areas being maybe 500 lumen / m2 and the sun being 10 000 lumen / m2)

This is what I would call physically correct output.

Reply:
Your incorrect assumption is that the eye handles over-bright spots using only the pupil (a linear scale factor akin to aperture). This is somewhat accurate if you focus on the bright spot, but not at all if the spot is out of your central vision. On a sunny day the sun can easily be "in your eyes" without your looking straight at it. However, if your pupil clamped down enough to resolve that 100,000 lux range, the rest of your scene would be black by comparison. Instead, your retina is also nonlinear, decreasing the response from those over-stimulated cells while still resolving the the rest of the scene at reasonable lux, hence your ability to see the horizon while the sun appears white (without looking straight at it). 

You're right that a LUT *can* be used for artistic intent, but its most common use is actually to make neutral product photography look right in sRGB or HDR (hence the IKEA beauty curve). It is is physically correct in the sense that it models part of the eye's physical response to stimulus, and it is necessary in making an image "photo-realistic". 

The difficulty with #2083 is that it specifies everything right down to the display without leaving a place for the LUT to be applied. You mention applying it as a post-process, but that's exactly what this extension does. Once you're down to PQ, there isn't anywhere left to apply a post-process, and you've already lost the information you need due to dropping from floating point to 10-bit.

Reply:
>You're right that a LUT can be used for artistic intent, but its most common use is actually to make neutral product photography look right in sRGB or HDR (hence the IKEA beauty curve).

What you call 'Ikea beauty curve' is applied to photographs as a last step just before publishing and shall not be part of the asset.
This type of color-grading (or tone-mapping) - if it needs to be applied to glTFs - belongs in the engine or viewer in my opinion.
Adding color-grading in this way to glTF assets is a slippery slope that opens up for a multitude of problems.

>The difficulty with https://github.com/KhronosGroup/glTF/pull/2083 is that it specifies everything right down to the display without leaving a place for the LUT to be applied.

Reluctant to discuss another extension here - at the same time I do not want misconceptions to get foothold.
No, that is clearly not what KHR_displaymapping_pq does.

For an overview of where color-grading _could_ be implemented together with KHR_displaymapping_pq - please look at the section about integration points and motivation.
This should give you the information needed - if not please let me know how I can clarify!